.github/action/action.yml

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020-2022 F4PGA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: "Run tests"
+
+inputs:
+  plugin:
+    description: "Name of the plugin to run the tests for"
+    required: true
+  yosys-version:
+    description: "Which yosys version to build against ('conda' or branch name)"
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+
+    - uses: actions/checkout@v2
```

Comment:
I think this is redundant in a local Composite Action. The Composite Action requires you to checkout the repository before, because you need to provide a local path to a file that must exist. Therefore, I recommend to add the `submodules: recursive` in the `tests.yml` workflow, and remove this step here.

Alternatively, you can use a Reusable Workflow, instead of a Composite Action. The syntax is quite similar, but you will need to locate the reusable workflow in `.github/workflows/`. Then, you can call the local reusable workflow without checking out the repo. So, you would remove the step from `tests.yml` and keep it here.

Furthermore, Reusable Workflows allow you to have multiple jobs in them, not only steps.

Reply:
I removed this checkout step from composite action and added `with: submodule: recursive' in `tests.yml`: https://github.com/chipsalliance/yosys-f4pga-plugins/pull/433/commits/64be765b636f0519c0c96b123d7a7a316c5c7837